build.zig.zon

```diff
@@ -0,0 +1,13 @@
+.{
+    .name = "diffz",
+    .version = "0.0.1",
+    .paths = .{
+        "DiffMatchPatch.zig",
+        ".gitattributes",
+        ".gitignore",
```

Comment:
Are you sure? The standard template for `build.zig.zon` says this: 

```
        // This makes *all* files, recursively, included in this package. It is generally
        // better to explicitly list the files and directories instead, to insure that
        // fetching from tarballs, file system paths, and version control all result
        // in the same contents hash.
```

The contents hash is calculated from the files listed in `zig.build.zon`, and I've found that leaving out files means that another package can't actually find the dependency.  I think these file names should be left in, and then removed if some integration testing shows that they don't have to be included.

Reply:
I am specifically talking about `.gitattributes` and `.gitignore`. A projects that dependents on this diffz project has not use for these files. This would also apply if there were a `tests` subdirectory. If you were to leave out the `build.zig` then obviously the package wouldn't work as dependency but I am just talking about these two files.

Reply:
Right, but if you take a tarball using `zig fetch --save`, then empirically, the build system won't successfully link to the dependency, because the hash it generates will be wrong.  I've seen this happen setting up another library to use `build.zig.zon`. 

So while I see your point about not having any use for those files, I think they need to stay for the current iteration of the build system to work successfully.  

It's worth checking this after the PR, I suppose. If I'm wrong, and the build failures I was seeing aren't actually correlated with my including files I didn't think needed to be included, it's very easy to remove them.

Reply:
tbh I have no clue what any of this has to do with the removal of `.gitattributes` and `.gitignore` from `paths`? Whatever issue you are trying to describe is unrelated to what I am talking about.

Here are some package that don't include these two files but still work:
- https://github.com/ziglibs/known-folders/blob/master/build.zig.zon
- https://github.com/zigtools/zig-lsp-codegen/blob/main/build.zig.zon
- https://github.com/hexops/mach/blob/main/build.zig.zon
- https://github.com/zigzap/zap/blob/master/build.zig.zon

Reply:
My stance on this is fairly simple: I wrote the `build.zig.zon` file in question, and I've applied it, by using `zig fetch --save` and building a program with it. 

The two file names you want to remove from the path, I specifically added, to get that to work. It's possible that I made other changes to the code and it's a coincidence, that's not how I remember it though.

The `build.zig.zon` file is new, so there's no risk of compatibility problems with an old file which didn't have it. If *you* can confirm that removing those file names still results in a working build, that's great, it's trivial to remove them. 

But I think that's how this should happen, if it does happen. Because I know for a fact that it's functional in its current state.

Reply:
> Whatever issue you are trying to describe is unrelated to what I am talking about.

I'll clarify: specifically, when fetching a `.tar.gz` from a release tag, if the tarball has different files from the `zig.build.zon` spec, the hash was wrong. I don't know whether or not that's true for fetching from a repo root directly, but it's important to support both cases, since `zig fetch` can do both of those things.  

I am happy to be wrong about that, but what I won't do is put additional time into confirming that those two path names have to be where they are. If you determine that isn't true, great, I have no qualms about removing them.

Reply:
btw, if you do a git blame on that file, you'll see that those two lines were added at a different time from the rest.

Reply:
> The two file names you want to remove from the path, I specifically added, to get that to work. It's possible that I made other changes to the code and it's a coincidence, that's not how I remember it though.

Could elaborate on that. What issue were you experiencing? I am very confused that you somehow had to add these files to `paths` in `build.zig.zon` to resolve it.

Were you for example failing to build a project that depends on 4792861d6d423f6a8fead9db6c7f0b0024b22136? (latest commit before you messed with `build.zig.zon`)

> If _you_ can confirm that removing those file names still results in a working build, that's great, it's trivial to remove them.

Just try this out! Create a new project with these three files and tell me if it doesn't compile. The diffz commit I am depending on is 4792861d6d423f6a8fead9db6c7f0b0024b22136  but any other should also work.

<details>

<summary>src/main.zig</summary>

```zig
const std = @import("std");
const diffz = @import("diffz");
pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const result = try diffz.default.diff(arena.allocator(), "foo", "boor", true);
    std.debug.print("{any}\n", .{result.items});
}
```

</details>

<details>

<summary>build.zig</summary>

```zig
const std = @import("std");

pub fn build(b: *std.Build) void {
    const target = b.standardTargetOptions(.{});
    const optimize = b.standardOptimizeOption(.{});

    const exe = b.addExecutable(.{
        .name = "zig-test",
        .root_source_file = b.path("src/main.zig"),
        .target = target,
        .optimize = optimize,
    });
    b.installArtifact(exe);

    const diffz_module = b.dependency("diffz", .{}).module("diffz");
    exe.root_module.addImport("diffz", diffz_module);

    const run_step = b.step("run", "Run main");
    run_step.dependOn(&b.addRunArtifact(exe).step);
}

```

</details>

<details>

<summary>build.zig.zon</summary>

```zig
.{
    .name = "zig-test",
    .version = "0.0.0",
    .dependencies = .{
        // Both of the result in the same hash
        // zig fetch --save git+https://github.com/ziglibs/diffz.git#4792861d6d423f6a8fead9db6c7f0b0024b22136
        // zig fetch --save https://github.com/ziglibs/diffz/archive/4792861d6d423f6a8fead9db6c7f0b0024b22136.tar.gz
        .diffz = .{
            .url = "git+https://github.com/ziglibs/diffz.git#4792861d6d423f6a8fead9db6c7f0b0024b22136",
            .hash = "1220acf95f2b526cf8238ab31aeebb398448c37b45e56530b537aebc5d7beb6715d1",
        },
    },
    .paths = .{"."},
}

```

</details>

<details>

If you were experiencing a hash mismatch error then you compare the hashes of dependencies with `zig fetch --debug-hash URL`

<summary>zig fetch --debug-hash</summary>

Only these 5 files need to exist and match the provided hash in `build.zig.zon`. All other files are filtered out and do not change the hash.

```bash
> zig fetch --debug-hash git+https://github.com/ziglibs/diffz.git#4792861d6d423f6a8fead9db6c7f0b0024b22136
file: 8d1801d14bfd92f9d094485c1f4b88e69f3ff3f0493ace9bd1f9216b0cf6558d: DiffMatchPatch.zig
file: 5a8291845ff2bfeef5134d7e73be726facca9d62b4be94bbd4677e3668ae0177: LICENSE
file: fccbc1f2528c298081b4a01d1387694be0f746ce463e86eafa84c2521e399d75: README.md
file: 4735f1724f564bfbc562e6888cf79b7975f4f37cf2125f89d6a4fd19035510a5: build.zig
file: 20edf3a874b5afdae3aa7f1724b9916c25d28de0f4bbf725b37d4d6d3c8ad545: build.zig.zon
1220acf95f2b526cf8238ab31aeebb398448c37b45e56530b537aebc5d7beb6715d
```

</details>